src/swupd/cmd/create-fullfiles/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+// Creates the fullfiles for a specific version given a state
+// directory. Used to test the logic in the library functions.
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime/pprof"
+
+	"swupd"
+)
+
+func usage() {
+	fmt.Fprintf(os.Stderr, `Usage: create-fullfiles [FLAGS] STATEDIR VERSION
+Flags:
+`)
+	flag.PrintDefaults()
+	os.Exit(2)
+}
+
+func main() {
+	log.SetFlags(0)
+
+	cpuProfile := flag.String("cpuprofile", "", "write CPU profile to a file")
+	outputDir := flag.String("o", "", "output directory, creates temporary if not specified")
+	flag.Usage = usage
+
+	if *cpuProfile != "" {
+		f, err := os.Create(*cpuProfile)
+		if err != nil {
+			log.Fatalf("couldn't create file for CPU profile: %s", err)
+		}
+		pprof.StartCPUProfile(f)
+		defer pprof.StopCPUProfile()
+	}
+
+	flag.Parse()
```

Comment:
Doesn't flag.Parse() need to go before line 33 to get *cpuProfile set?

Reply:
Last minute copy pasta. Will fix this.